### PR TITLE
HP-685: Local Canonical Token Lists

### PR DESCRIPTION
**Pull request by nabsta-eth**

### CODE REVIEW

**Comment by rellfy** [comment]
Some tests are failing

File: src/types/Convert.ts [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+import polygon from "../../tokens/polygon.json";
 
 type Token = {
   symbol: string;
+  name: string;
   address: string;
   decimals: number;
+  icon: string;
+  highlight: boolean;
+  displayDecimals: number;
```
Comment (rellfy) [inline src/types/Convert.ts:13]: What are these for?

Comment (rellfy) [reply src/types/Convert.ts:13]: We could possibly get rid of this type here and implement the other [Token](https://github.com/handle-fi/handle-sdk/blob/master/src/types/Token.ts) type instead.

Comment (nabsta-eth) [reply src/types/Convert.ts:13]: - `highlight` is tokens to be shown in the dropdown (pre-search);
- `displayDecimals` is used for showing display amounts.
Their addition avoids having to "map" through the whole list (e.g. 1138 for mainnet) and add them on load to make convert "work".

Comment (ahinchliff) [reply src/types/Convert.ts:13]: I think highlight is a client responsibility.

**Review by ahinchliff — APPROVED** [review]

File: src/types/Convert.ts (hunk 1 above)
Comment (rellfy) [reply src/types/Convert.ts:13]: I think both are client parameters, we don't need them in the SDK

Comment (rellfy) [reply src/types/Convert.ts:13]: I think we should remove highlight, I would also remove displayDecimals but if it's convenient then cool

File: src/types/Convert.ts [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+import homestead from "../../tokens/homestead.json";
+import polygon from "../../tokens/polygon.json";
 
 type Token = {
   symbol: string;
+  name: string;
   address: string;
   decimals: number;
+  icon: string;
+  highlight: boolean;
```
Comment (rellfy) [inline src/types/Convert.ts:12]:
```suggestion
```

**Review by rellfy — CHANGES_REQUESTED** [review]

**Comment by rellfy** [comment]
If all the display decimals are 4, does it have to be a value per-token? Can't it be a constant set from the vue app?

**Pull request merged**